JsonLogic/Rules/AllRule.cs

```diff
@@ -57,11 +57,11 @@ protected internal AllRule(Rule input, Rule rule)
 	}
 }
 
-internal class AllRuleJsonConverter : AotCompatibleJsonConverter<AllRule>
+internal class AllRuleJsonConverter : WeaklyTypedJsonConverter<AllRule>
 {
 	public override AllRule? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
 	{
-		var parameters = options.Read(ref reader, LogicSerializerContext.Default.RuleArray);
+		var parameters = options.Read(ref reader, JsonLogicSerializerContext.Default.RuleArray);
 
```

Comment:
These should switch to options.ReadArray. It's probably fine because we don't have any tests that are trying to pass in an options dictionary that lacks the Rule type so we aren't tripping the bug in all these rule converters.